src/units/filter.cpp

```diff
@@ -666,8 +666,15 @@ void unit_filter_compound::fill(vconfig cfg)
 		create_attribute(literal["formula"],
 			[](const config::attribute_value& c)
 			{
-				//TODO: catch syntax error.
-				return wfl::formula(c, new wfl::gamestate_function_symbol_table());
+				try {
+					return wfl::formula(c, new wfl::gamestate_function_symbol_table());
+				} catch(const wfl::formula_error& e) {
+					lg::log_to_chat() << "Formula error while evaluating formula in unit filter, reset to empty: " << e.type << " at "
+									  << e.filename << ':' << e.line << ")\n";
+					ERR_WML << "Formula error while evaluating formula in unit filter, reset to empty: " << e.type << " at "
+							<< e.filename << ':' << e.line << ")";
+					return wfl::formula("", new wfl::gamestate_function_symbol_table());
```

Comment:
Why does this say "reset to empty"? I get that you're returning an "empty" formula as part of the error, but that doesn't seem like something worth stating in the error message.

(Is returning an empty formula fine, by the way?)

Reply:
> In my test, weapon special with empty formula seems to be ignored currently, only err msg like "Pango-critical" come out. Besides, if the error is caught here, it has to return a value. Considering the ignoring behavior of empty formula, I can't think of a better value than empty personally.

It's a little tricky since it returns a formula rather than a value, but I think `"null()"` might be appropriate here.

Reply:
Also, there's no point in adding `new wfl::gamestate_function_symbol_table()` here when the formula is a literal string that doesn't even need that function table. I think the function table argument is optional. If not, I'm pretty sure you can pass `nullptr`.